backend/app/api/ai_agent.py

```
@@ -0,0 +1,288 @@
+
+
+# ── 4.2 GET /decisions (cursor pagination) ────────────────────────────────────
+
+
+@router.get(
+    "/decisions",
+    response_model=DecisionListOut,
+    dependencies=[Depends(get_current_user)],
+)
+async def list_decisions(
+    cursor: datetime | None = Query(default=None, description="timestamp < cursor (ISO8601)"),
```

Ruff (0.15.10)

[warning] Do not perform function call `Query` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)